### PR TITLE
fix: .env Namespaced Character Secrets

### PR DESCRIPTION
**Pull request by genecyber**
# Relates to:
N/A

# Risks
Medium - This PR adds a new feature for character-specific environment variable namespacing.

Risks include:
- Changes to environment variable handling
- Changes to character settings loading
- Potential impact on existing character configurations

# Background
I wanted to be able to check in character files without having to sanitize them first

### Feature Implementation
Adds support for character-specific namespaced environment variables:
1. New environment variable pattern: `CHARACTER.YOUR_CHARACTER_NAME.SETTING_NAME`
2. Automatic handling of spaces in character names (converted to underscores)
3. Enhanced settings hierarchy:
   - Character-specific namespaced env variables (highest priority)
   - Character settings from JSON
   - Global environment variables
   - Default values
4. Maintains backward compatibility with existing settings system

Key changes:
- Modified settings loader in `packages/core/src/settings.ts`
- Updated character loading in `packages/agent/src/index.ts`
- Added automatic conversion of character name spaces to underscores

### Documentation Updates
Updated documentation to reflect the new feature:
1. Added namespaced settings pattern to secrets management guide
2. Updated configuration guide with new examples
3. Updated agent package documentation with hierarchy explanation
4. Added clear examples for both .env and character.json methods

## What kind of change is this?
Features (non-breaking change which adds functionality)

# Documentation changes needed?
Yes - Documentation has been updated to reflect the new feature:
- Added namespaced settings pattern
- Updated configuration examples
- Updated settings hierarchy explanation
- Added character name handling instructions

# Testing

### Feature Testing
1. Review code changes:
   - `packages/core/src/settings.ts`
   - `packages/agent/src/index.ts`

2. Test functionality:
   ```env
   # Test with simple name
   CHARACTER.TESTBOT.OPENAI_API_KEY=sk-test
   
   # Test with spaces in name
   CHARACTER.MY_TEST_BOT.ANTHROPIC_API_KEY=sk-other
   ```

### Documentation Testing
Review updated docs:
- docs/docs/guides/secrets-management.md
- docs/docs/guides/configuration.md
- docs/docs/packages/agent.md

## Detailed testing steps

1. Feature Testing:
   - Create character with spaces in name
   - Add namespaced variables to .env
   - Verify correct loading into character settings
   - Test precedence over character.json settings
   - Verify backward compatibility
   - Test global fallback behavior

2. Documentation Verification:
   - Verify pattern examples are correct
   - Verify hierarchy explanation is clear
   - Verify character name handling is explained
   - Test example configurations

# Screenshots
N/A - Feature implementation and documentation changes

### CODE REVIEW

File: docs/docs/guides/secrets-management.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ A comprehensive guide for managing secrets, API keys, and sensitive configuratio
 
 Eliza uses a hierarchical environment variable system:
 
-1. Character-specific secrets (highest priority)
+1. Character-specific namespaced environment variables (highest priority)
+1. Character-specific secrets
```
Comment (monilpat) [inline docs/docs/guides/secrets-management.md:16]: nit: numbering

**Comment by monilpat** [comment]
Thanks for getting this out! Great idea - can we add a quick unit test to confirm that the cascading logic (in terms of where to read the settings) works as expected. If you don't want to include this in this PR no worries create a ticket and TODO comment. But please add a screengrab or confirm on your machine that the precedence is honored re:

**Comment by genecyber** [comment]
good idea, I'll add tests. Is the suite in pretty good condition? I didn't try yet.

**Comment by odilitime** [comment]
replaced by #1454 which fixed conflicts

**Pull request closed without merging**